### PR TITLE
feat(oas) - accurate model OAS representation - F to O

### PR DESCRIPTION
**Pull request by patrick-medusajs**
### Scope

Models F to O

### What

Refactor OAS for models to accurately represent their shape in API responses.

### Why

About 33% of model fields are not accurately represented in the OAS. Most of the issues are:
- fields that can not be omitted in the response are not declared as `required`
- fields that could return `null` as their value are not declared as `nullable: true`

When using a code generator, these OAS issues would lead to inaccurate response shapes in the generated client.

#### nullable
Fields meeting at least one of the following condition will be represented as `nullable: true` in OAS:
* The field is decorated with `@Column({ nullable: true })`
* The field is decorated with `@OneToOne`, `@ManyToOne`
* The field is decorated with `@DeleteDateColumn`

#### optional
Fields meeting at least one of the following conditions will never be listed as `required` in OAS and will be considered optional and could be omitted in the response:
* The field is decorated with `@OneToOne`, `@ManyToOne`, `@OneToMany`, `@ManyToMany`
* The field is decorated with `@FeatureFlagColumn`
* The field is decorated with `@Column({select: false})`
* The field is representing dynamic values not persisted in the database

Fields not meeting any of the conditions above will be declared as `required` and are expected to be present in the response.

### Test
* Ran OAS validator.
* Ran docs build script.

Expect OAS changes to be reflected in the API documentation.

### CODE REVIEW

File: packages/medusa/src/models/notification.ts [hunk 1]
```diff
@@ -70,120 +70,81 @@ export class Notification extends BaseEntity {
...
- *     description: "The date with timezone at which the resource was created."
- *     format: date-time
- *   updated_at:
- *     type: string
- *     description: "The date with timezone at which the resource was updated."
- *     format: date-time
- */
-
-/**
- * @schema NotificationResend
```
Comment (patrick-medusajs) [inline packages/medusa/src/models/notification.ts:131]: NotificationResend doesn't exist in the codebase.

**Review by patrick-medusajs — COMMENTED** [review]
self-review: I'd recommend using the [Model TS refactor PR](https://github.com/medusajs/medusa/pull/3202/files) to facilitate the review process. I've also create a [demo PR](https://github.com/medusajs/medusa/pull/3204/files) to showcase the generate types from the OAS.

File: packages/medusa/src/models/invite.ts [hunk 2]
```diff
@@ -43,49 +43,60 @@ export class Invite extends SoftDeletableEntity {
...
  *   id:
  *     type: string
  *     description: The invite's ID
  *     example: invite_01G8TKE4XYCTHSCK2GDEP47RE1
  *   user_email:
- *     type: string
  *     description: The email of the user being invited.
+ *     type: string
  *     format: email
  *   role:
```
Comment (shahednasser) [inline packages/medusa/src/models/invite.ts:65]: This should be `nullable`

File: packages/medusa/src/models/line-item.ts [hunk 3]
```diff
@@ -207,120 +236,133 @@ export class LineItem extends BaseEntity {
...
+ *     description: Flag to indicate if new Line Items with the same variant should be merged or added as an additional Line Item.
  *     type: boolean
- *     example: false
+ *     default: true
  *   allow_discounts:
- *     description: "Flag to indicate if the Line Item should be included when doing discount calculations."
+ *     description: Flag to indicate if the Line Item should be included when doing discount calculations.
  *     type: boolean
- *     example: false
+ *     default: false
```
Comment (shahednasser) [inline packages/medusa/src/models/line-item.ts:281]: default should be `true`.

**Review by shahednasser — CHANGES_REQUESTED** [review]

File: packages/medusa/src/models/line-item.ts (hunk 3 above)
Comment (patrick-medusajs) [reply packages/medusa/src/models/line-item.ts]: Good catch. Done in e31492da48e1d04b745c5c39e465fb9583694a63

File: packages/medusa/src/models/invite.ts (hunk 2 above)
Comment (shahednasser) [reply packages/medusa/src/models/invite.ts:65]: yes I meant role, sorry for not being clear about it!

**Review by shahednasser — APPROVED** [review]
LGTM

**Pull request merged**